EIPS/eip-3540.md

```diff
@@ -93,88 +90,95 @@ The list of section headers is terminated with the *section headers terminator b
+| magic                  | 2 bytes  | 0xEF00        |                                                                                                              |
+| version                | 1 byte   | 0x01          | EOF version                                                                                                  |
+| kind_type              | 1 byte   | 0x01          | kind marker for type section                                                                                 |
+| type_size              | 2 bytes  | 0x0004-0x1000 | 16-bit unsigned big-endian integer denoting the length of the type section content, 4 bytes per code section |
+| kind_code              | 1 byte   | 0x02          | kind marker for code size section                                                                            |
+| num_code_sections      | 2 bytes  | 0x0001-0x0400 | 16-bit unsigned big-endian integer denoting the number of the code sections                                  |
+| code_size              | 2 bytes  | 0x0001-0xFFFF | 16-bit unsigned big-endian integer denoting the length of the code section content                           |
+| kind_container         | 1 byte   | 0x03          | kind marker for container size section                                                                       |
+| num_container_sections | 2 bytes  | 0x0001-0x00FF | 16-bit unsigned big-endian integer denoting the number of the container sections                             |
+| container_size         | 2 bytes  | 0x0001-0xFFFF | 16-bit unsigned big-endian integer denoting the length of the container section content                      |
+| kind_data              | 1 byte   | 0x04          | kind marker for data size section                                                                            |
+| data_size              | 2 bytes  | 0x0000-0xFFFF | 16-bit unsigned big-endian integer denoting the length of the data section content (*)                       |
+| terminator             | 1 byte   | 0x00          | marks the end of the header                                                                                  |
+
+(*) For not yet deployed containers this can be greater than the actual content length.
+
+#### Body
+
+| name              | length   | value         | description                                                                                |
+|-------------------|----------|---------------|--------------------------------------------------------------------------------------------|
+| types_section     | variable | n/a           | stores code section metadata                                                               |
+| inputs            | 1 byte   | 0x00-0x7F     | number of stack elements the code section consumes                                         |
+| outputs           | 1 byte   | 0x00-0x80     | number of stack elements the code section returns or 0x80 for non-returning functions      |
```

Comment:
I'm not a fan of this, because without it it seems you can implement the format according to EIP and then pass 0 tests.

Maybe I'm wrong, because you wouldn't pass anything without implementing the rest of validation anyway.

Reply:
Following the convention we seem to have adopted recently with the other EIPs, I'll leave a note that one of `outputs`' special values has already been assigned in a separate EIPs, or sth like that.